scripts/tidy.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+
+# this is for developer local check only.
+# not used by github action
+
+if [ -z "${MLIR_DIR}" ]; then
+    echo "The environment variable MLIR_DIR is not set."
+    exit 1
+fi
+
+# set PR_REF if your pr target is not main
+PR_REF=${PR_REF:-main}
+
+MERGE_BASE=$(git merge-base $PR_REF HEAD)
+CHANGED_FILES=$(git diff --name-only $MERGE_BASE HEAD | paste -sd' ')
+
+mkdir -p build
+cd build
```

Comment:
Maybe use the build dir in the project root? 
```suggestion
BUILD_DIR="$(dirname "$0")/../build/tidy"
mkdir -p "$BUILD_DIR"
cd "$BUILD_DIR"
```

Reply:
@AndreyPavlenko 
It is the same problem mentioned in https://github.com/intel/graph-compiler/issues/93#issuecomment-2124141173

It is not ensured that developer is using the clang compiler, we cannot reuse that folder.